heartbeat/tests/system/test_monitor.py

```diff
@@ -41,34 +42,39 @@ def test_http(self, status_code):
         self.assert_fields_are_documented(output[0])
 
     @parameterized.expand([
-        ("8185", "up"),
-        ("8186", "down"),
+        (lambda server: "localhost:{}".format(server.server_port), "up"),
+        # This IP is reserved in IPv4
+        (lambda server: "207.254.52.220:1233", "down"),
     ])
-    def test_tcp(self, port, status):
+    def test_tcp(self, url, status):
         """
         Test tcp server
         """
         server = self.start_server("hello world", 200)
-        self.render_config_template(
-            monitors=[{
-                "type": "tcp",
-                "hosts": ["localhost:" + port],
-            }],
-        )
+        try:
+            self.render_config_template(
+                monitors=[{
+                    "type": "tcp",
+                    "hosts": [url(server)],
+                    "timeout": "3s"
+                }],
+            )
 
-        proc = self.start_beat()
-        self.wait_until(lambda: self.log_contains("heartbeat is running"))
+            proc = self.start_beat()
+            try:
+                self.wait_until(lambda: self.log_contains(
+                    "heartbeat is running"))
 
-        self.wait_until(
-            lambda: self.output_has(lines=1))
+                self.wait_until(
+                    lambda: self.output_has(lines=1))
+            finally:
+                proc.check_kill_and_wait()
 
-        proc.check_kill_and_wait()
-
-        server.shutdown()
-
-        output = self.read_output()
-        assert status == output[0]["monitor.status"]
-        if os.name == "nt":
-            # Currently skipped on Windows as fields.yml not generated
-            raise SkipTest
-        self.assert_fields_are_documented(output[0])
+            output = self.read_output()
+            nose.tools.assert_equal(status, output[0]["monitor.status"])
```

Comment:
Could we use `assert_last_status` here?
```suggestion
            self.assert_last_status(status)
```